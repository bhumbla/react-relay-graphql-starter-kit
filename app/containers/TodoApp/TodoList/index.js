/* @flow */

import React from 'react';
import Relay from 'react-relay';
import TodoItem from '../TodoItem';

type TodoEdge = {
  node: {
    id: string,
  },
};

type Props = {
  todos: {
    edges: Array<TodoEdge>,
  },
  user: Object
};

/**
 * Todo list component
 * Parent component: TodoApp
 * Child component: TodoItem
 */
class TodoList extends React.Component {

  props: Props;

  render() {
    return (
      <div>
        <h2>Todo List</h2>
        {this.props.todos.edges.map(edge =>
          <TodoItem
            key={edge.node.id}
            todo={edge.node}
            user={this.props.user}
          />,
        )}
      </div>
    );
  }
}

export default Relay.createContainer(TodoList, {
  fragments: {
    todos: () => Relay.QL`
      fragment on TodoConnection {
        edges {
          node {
            id
            ${TodoItem.getFragment('todo')}
          }
        }
      }
    `,
    user: () => Relay.QL`
      fragment on User {
        ${TodoItem.getFragment('user')}
      }
    `,
  },
});
