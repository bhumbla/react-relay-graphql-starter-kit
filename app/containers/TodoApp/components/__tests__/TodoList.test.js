import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { graphql } from 'graphql';
import schemaString from 'data/schema.graphql';
import { TodoList } from '../TodoList';

global.React = React;

const schema = makeExecutableSchema({ typeDefs: schemaString });

// Add mocks, modifies schema in place
addMockFunctionsToSchema({ schema });

const query = `
query tasksForUser {
  user {
    id
    todos(first: 10) {
      edges {
        node {
          id
          content
        }
      }
    }
  }
}
`;

describe('TodoList component', () => {
  let todoList;

  const todos = {
    edges: [
      {
        node: {
          id: '1',
        },
      },
      {
        node: {
          id: '2',
        },
      },
    ],
  };

  const user = {};

  describe('Tests with undefined todos and user props', () => {
    beforeEach(() => {
      todoList = shallow(<TodoList todos={todos} user={user} />);
    });

    it('Renders correctly without todos and user props', () => {
      expect(shallowToJson(todoList)).toMatchSnapshot();
      graphql(schema, query).then(result => console.log('Got result', result, result.data.user.todos.edges)); // eslint-disable-line
    });
  });
});
