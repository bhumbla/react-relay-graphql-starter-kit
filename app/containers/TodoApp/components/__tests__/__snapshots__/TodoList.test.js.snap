exports[`TodoList component Tests with undefined todos and user props Renders correctly without todos and user props 1`] = `
<div
  className="todo-list-component">
  <Relay(TodoItem)
    todo={
      Object {
        "id": "1",
      }
    }
    user={Object {}} />
  <Relay(TodoItem)
    todo={
      Object {
        "id": "2",
      }
    }
    user={Object {}} />
</div>
`;
